File: ssluzba konacno/src/main/java/global/LoginPage.java
package global;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

import helpers.Utils;

public class LoginPage {
	private WebDriver driver;

	public LoginPage(WebDriver driver) {
		this.driver = driver;
	}
	
	
	
	public WebElement pocetnaProvera() {
		return Utils.waitForElementPresence(driver, By.xpath("//p [contains(text(), 'Prijavite se da počnete sa radom.')]"), 15);
	}
	
	

	public WebElement getUsername() {
		return Utils.waitForElementPresence(driver, By.id("username"), 10);
	}

	public void setUsername(String username) {
		WebElement usernameInput = getUsername();
		usernameInput.clear();
		usernameInput.sendKeys(username);
	}
	
	public WebElement usernameRequiredError() {
		return Utils.waitForElementPresence(driver, By.xpath("//span [contains(text(), 'Korisničko ime obavezno')]"), 15);
	}
	

	public WebElement getPassword() {
		return Utils.waitForElementPresence(driver, By.id("password"), 10);
	}
	public void setPassword(String password) {
		WebElement passwordInput = getPassword();
		passwordInput.clear();
		passwordInput.sendKeys(password);
	}
	public WebElement passwordRequiredError() {
		return Utils.waitForElementPresence(driver, By.xpath("//span [contains(text(), 'Lozinka obavezna')]"), 15);
	}
	
	
	public WebElement errorInLogin() {
		return Utils.waitForElementPresence(driver, By.xpath("//li [contains(text(), 'Pogrešno korisničko ime ili lozinka!')]"), 15);
	}
	public WebElement errorInLogin2() {
		return Utils.waitForElementPresence(driver, By.xpath("//li [contains(text(), 'Problem sa logovanjem')]"), 15);
	}
	
	 

	public WebElement getPrijaviSeBtn() {
		return Utils.waitForElementPresence(driver, By.xpath("//button[@type='submit']"), 10);
	}

	public WebElement getOdustaniBtn() {
		return Utils.waitForElementPresence(driver, By.xpath("//button[@type='button']"), 10);
	}
	

	public void login(String username, String password) {
		setUsername(username);
		setPassword(password);
		getPrijaviSeBtn().click();
	}

}

File: ssluzba konacno/src/test/javascript/e2e/pages/globalzavrsni/login.page.js
var LoginPage = function() {};
var utils = require('../utils.js');


LoginPage.prototype = Object.create({}, {
    
   
    username: {
        get: function() {
            return utils.waitForElementPresence(by.id('username'), 10000);
        },
        set: function(value) {
            this.username.clear();
            this.username.sendKeys(value);
        }
    },
    usernameRequired: {
        get: function() {
            return utils.waitForElementPresence(by.xpath("//span [contains(text(), 'Korisničko ime obavezno')]"), 10000);
        }
    },
    
    
    password: {
        get: function() {
            return utils.waitForElementPresence(by.id('password'), 10000);
        },
        set: function(value) {
            this.password.clear();
            this.password.sendKeys(value);
        }
    },
    passwordRequired: {
        get: function() {
            return utils.waitForElementPresence(by.xpath("//span [contains(text(), 'Lozinka obavezna')]"), 10000);
        }
    },
    
    
    prijaviSeBtn: {
        get: function() {
            return utils.waitForElementPresence(by.buttonText('Prijavi se'), 10000);
        }
    },

    OdustaniBtn: {
        get: function() {
            return utils.waitForElementPresence(by.xpath("//button[@type='button']"), 10000);
        }
    },
    
    loginError: {
        get: function() {
            return utils.waitForElementPresence(by.xpath("//li[contains (text(),'Pogrešno korisničko ime ili lozinka!')]"), 10000);
        }
    },
    loginError1: {
        get: function() {
            return utils.waitForElementPresence(by.xpath("//span[contains (text(),'Problem sa logovanjem')]"), 10000);
        }
    },
    
  
    login: {
        value: function(usernameString, passwordString) {
            this.username = usernameString;
            this.password = passwordString;
            this.prijaviSeBtn.click();
        }
    }
});

// Export klase
module.exports = LoginPage;